osu.Game/Graphics/Containers/FacadeContainer.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.MathUtils;
+using osu.Game.Screens.Menu;
+using osuTK;
+
+namespace osu.Game.Graphics.Containers
+{
+    /// <summary>
+    /// A container that creates a <see cref="Facade"/> to be used by its children.
+    /// This container also updates the position and size of the Facade, and contains logic for tracking an <see cref="OsuLogo"/> on the Facade's position.
+    /// </summary>
+    public class FacadeContainer : Container
+    {
+        protected virtual Facade CreateFacade() => new Facade();
+
+        public Facade Facade { get; }
+
+        /// <summary>
+        /// Whether or not the logo assigned to this FacadeContainer should be tracking the position its facade.
+        /// </summary>
+        public bool Tracking;
+
+        private OsuLogo logo;
+        private float facadeScale;
+        private Vector2 startPosition;
+        private Easing easing;
+        private double startTime;
+        private double duration;
+
+        public FacadeContainer()
+        {
+            Facade = CreateFacade();
+        }
+
+        /// <summary>
+        /// Assign the logo that should track the Facade's position, as well as how it should transform to its initial position.
+        /// </summary>
+        /// <param name="logo"> The instance of the logo to be used for tracking. </param>
+        /// <param name="facadeScale"> The scale of the facade. </param>
+        /// <param name="duration"> The duration of the initial transform. Default is instant.</param>
+        /// <param name="easing"> The easing type of the initial transform. </param>
+        public void SetLogo(OsuLogo logo, float facadeScale, double duration = 0, Easing easing = Easing.None)
+        {
+            if (logo != null)
+            {
+                this.logo = logo;
+            }
+
+            this.facadeScale = facadeScale;
+            this.duration = duration;
+            this.easing = easing;
+        }
+
+        private Vector2 logoTrackingPosition => logo.Parent.ToLocalSpace(Facade.ScreenSpaceDrawQuad.Centre);
+
+        protected override void UpdateAfterChildren()
+        {
+            base.UpdateAfterChildren();
+
+            if (logo == null || !Tracking)
+                return;
+
+            Facade.Size = new Vector2(logo.SizeForFlow * facadeScale);
+
+            if (Facade.IsLoaded && logo.Position != logoTrackingPosition)
```

Comment:
how can we get here without the facade being loaded?

Reply:
this is actually more of a check of whether or not the Facade has been added to the Draw hierarchy yet, will change to a more appropriate check.

Reply:
Because FacadeContainer is not directly adding facade to the draw hierarchy, it needs to wait until whatever component needs it has finished loading and has done so, which is why this check is here.

Reply:
A safer check would be checking it has a parent, IMO. Maybe add tests for the case of the facade being removed from a container and switching between containers (I believe this is currently untested).

Reply:
sure, can agree with that.